### PR TITLE
[PLT-4951] Don't send notifications if thread is showing in RHS

### PR DESCRIPTION
**Pull request by krjn**
If MM is focused and the notification is a reply to a thread
that is being shown in the RHS, then ignore it even if it's
in another channel - and don't make it bold (unread).

https://mattermost.atlassian.net/browse/PLT-4951

Fixes https://github.com/mattermost/mattermost-server/issues/7718

### CODE REVIEW

File: actions/notification_actions.test.jsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                team_role: {
+                    permissions: [],
+                },
+                channel_role: {
+                    permissions: [],
+                },
+            },
+        },
+        general: {
+            license: {IsLicensed: 'false'},
```
Comment (crspeller) [inline actions/notification_actions.test.jsx]: Lots of extra stuff in here. Is this all necessary?

Comment (krjn) [reply actions/notification_actions.test.jsx]: IIRC, no. I removed and then went back on it, considering more involved test cases could make use of it. I'm impartial about this - I can remove some of it.

Comment (crspeller) [reply actions/notification_actions.test.jsx]: Ya I would rather have it focused on exactly what the test needs.

Comment (krjn) [reply actions/notification_actions.test.jsx]: Done.

**Comment by hmhealey** [comment]
@esethna could confirm, but I don't think we care about push notifications here. This is extending the behaviour where you don't receive desktop notifications for a channel that you're viewing, but we don't support any behaviour like that for push notifications. Push notifications aren't sent by default when you're online, so they don't run into this same issue where you receive notifications when you're actively viewing a channel

**Comment by lieut-data** [comment]
Sounds good! Thanks, @hmhealey :)

**Comment by lieut-data** [comment]
@krjn, might you be able to prefer merging `origin/master` instead of force pushing? It's much easier to review the incremental changes from a merge than rewriting the PR history altogether. Take a look at https://developers.mattermost.com/blog/submitting-great-prs/#4-avoid-force-pushing for more details :)

**Comment by krjn** [comment]
> @krjn, might you be able to prefer merging `origin/master` instead of force pushing?

Hmm, I don't mind and when worthy to keep history I usually do but, in this case, there was no evolution of implementation and the rebase had no conflicts. I took the commits back - if preferred I can take it all back and merge.

**Comment by hanzei** [comment]
Thank you very much for this awesome contribution @krjn :tada:

**Comment by esethna** [comment]
Hi @krjn, firstly we super appreciate the contribution of this feature. However, upon reviewing with our QA and dev team internally and testing this on our community server we found some issues with the client side approach for this feature:

1. Since the messages are not marked as read on the server when they appear in a thread you have open, a refresh of the page causes the unread or mention badge that had previously been hidden to reappear. This can be a confusing UX
2. Say I have a thread for Channel A open in the RHS and Channel B open in the center pane. Then a new message is posted in Channel A, then a reply is made on the thread I have open from Channel A. Now I switch back to channel A and see the new messages line above both messages, even though I've already seen the one in the thread. This can be a confusing UX, especially in the case of mentions where the mention badge in the sidebar does not match the number of mentions shown under the new messages line.

Given the above UX concerns, we think it's best to revert this change until we can figure out an implementation method and UX that doesn't surface these issues. Are you okay with that?

**Comment by krjn** [comment]
Sure. Have any ideas been brought forward that I could maybe help explore? The only one I had was to mark as read ***if*** the only messages in the channel were from the thread.

**Pull request merged**